apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -0,0 +1,170 @@
+                            new SpannerParameter("Int64ArrayValue", SpannerDbType.ArrayOf(SpannerDbType.Int64), null),
+                            new SpannerParameter("Float64ArrayValue", SpannerDbType.ArrayOf(SpannerDbType.Float64), null),
+                            new SpannerParameter("StringArrayValue", SpannerDbType.ArrayOf(SpannerDbType.String), null),
+                            new SpannerParameter("BytesArrayValue", SpannerDbType.ArrayOf(SpannerDbType.Bytes), null),
+                            new SpannerParameter("TimestampArrayValue", SpannerDbType.ArrayOf(SpannerDbType.Timestamp), null),
+                            new SpannerParameter("DateArrayValue", SpannerDbType.ArrayOf(SpannerDbType.Date), null),
+                        })
+                        == 1);
+            using (var reader = await GetLastRowAsync())
+            {
+                readData = true;
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("BoolValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("Int64Value")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("Float64Value")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("StringValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("TimestampValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("DateValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("BoolArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("Int64ArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("Float64ArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("StringArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("BytesArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("TimestampArrayValue")));
+                Assert.True(null == reader.GetValue(reader.GetOrdinal("DateArrayValue")));
+            }
+            Assert.True(readData);
+        }
+
+        [Fact]
+        public async Task WriteValues()
+        {
+            bool readData;
+            var jBd = new DateTime(2017, 3, 17, 15, 30, 0);
+            var sBd = new DateTime(2017, 5, 9);
+            bool?[] bArray = { true, null, false};
+            long?[] lArray = {0, null, 1};
+            double?[] dArray = {0.0, null, 2.0};
+            string[] sArray = {"abc", null, "123"};
+            string[] bArrayArray =
+            {
+                Convert.ToBase64String(new byte[] {0, 1, 2}), null,
+                Convert.ToBase64String(new byte[] {1, 2, 3}),
+            };
+            DateTime?[] dtArray = new DateTime?[]
+            {
+                new DateTime(2017, 3, 17),
+                null,
+                new DateTime(2017, 5, 9), 
+            };
+            DateTime?[] tmArray = new DateTime?[]
```

Comment:
On the receiving side, I only use the built in converter between Timestamp (the proto type) and DateTime.  I'm assuming thats always UTC right?
On the way out, I use xmlconvert to get an rfc 3339 string -- so its on the caller to send me a datetime as appropriate (if its local time, it should still create the appropriate rfc 3339).